lib/system/qnx/io.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2025, QNX Software Systems Limited
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/*
+ * @file	qnx/io.c
+ * @brief	QNX libmetal io operations.
+ */
+
+#include <metal/io.h>
+#include <metal/sys.h>
+
+int metal_sys_io_mem_map(struct metal_io_region *io)
+{
+  if (io->virt != NULL)
+    return 0;
+	void *addr;
+  int flags, prot;
+
+  flags = (io->mem_flags & ~PROT_MASK) | MAP_PHYS;
+  prot = (io->mem_flags & PROT_MASK) | PROT_READ ;
```

Comment:
`PROT_WRITE` seems to have been removed.
```suggestion
  prot = (io->mem_flags & PROT_MASK) | PROT_READ | PROT_WRITE;
```